src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/PackageWriterImpl.java

```diff
@@ -235,6 +245,49 @@ public void addClassesSummary(SortedSet<TypeElement> classes, String label,
         }
     }
 
+    public void addPackageSummary(List<PackageElement> packages, Content label,
+                                  TableHeader tableHeader, Content summaryContentTree,
+                                  boolean showModules) {
+        if(!packages.isEmpty()) {
+            Table table = new Table(HtmlStyle.summaryTable)
+                    .setCaption(label)
+                    .setHeader(tableHeader);
+            if (showModules) {
+                table.setColumnStyles(HtmlStyle.colFirst, HtmlStyle.colExtra, HtmlStyle.colLast);
+            } else {
+                table.setColumnStyles(HtmlStyle.colFirst, HtmlStyle.colLast);
+            }
+
+            for (PackageElement pkg : packages) {
+                Content packageLink = getPackageLink(pkg, Text.of(pkg.getQualifiedName()));
+                Content moduleLink = HtmlTree.EMPTY;
+                if (showModules) {
+                    ModuleElement module = (ModuleElement) pkg.getEnclosingElement();
+                    if (module != null && !module.isUnnamed()) {
+                        moduleLink = getModuleLink(module, Text.of(module.getQualifiedName()));
+                    }
+                }
+                ContentBuilder description = new ContentBuilder();
+                addPreviewSummary(pkg, description);
+                if (utils.isDeprecated(pkg)) {
+                    description.add(getDeprecatedPhrase(pkg));
+                    List<? extends DeprecatedTree> tags = utils.getDeprecatedTrees(pkg);
+                    if (!tags.isEmpty()) {
+                        addSummaryDeprecatedComment(pkg, tags.get(0), description);
+                    }
+                } else {
+                    addSummaryComment(pkg, description);
+                }
+                if (showModules) {
+                    table.addRow(packageLink, moduleLink, description);
```

Comment:
This looks like the companion row to the header entry, to put the module column first.